### ISSUE TITLE
Feature Request: Display remote image, with polling refresh

### ISSUE THREAD

**Issue by donatellosantoro** [GeekMagicClock/smalltv-pro#42]
Hi @GeekMagicClock,
the SmallTV-Pro is really a small gem for a nerd! And thanks for being active in this community!

I understand that releasing the code of the firmware could be a threat for the commercialization of your product, and that implementing an SDK a hard work (but I really hope you'll be able to do so!)

As you see, many other users have nice ideas, and it would be difficult to implement all of them. So my idea is to move the application logic of some applications out of the SmallTV. 

In other words, it could be relatively simple for you to create a generic application that downloads, at a custom rate (e.g. 1 sec), at a specific URL, an IMAGE in 240x240.
So a developer could write in the server the logic of generating dynamically such images (e.g. by displaying a text, a complex small dashboard, a new weather station, and so on).

**One app to rule them all!**

What do you think?

**Comment by CezarMihnea** [comment]
This sounds really nice! Great idea. Similar to casting a screen, but from a Webserver custom programmed.

**Comment by donatellosantoro** [comment]
To stimulate the conversation, I have prepared a simple Dockerfile with a very basic app that returns a 240x240 image of the current weather.

https://github.com/donatellosantoro/smalltv-pro-webapp

![Alt text](https://github.com/donatellosantoro/smalltv-pro-webapp/raw/main/weather.png "Weather app")


Obviously, there are many different ways to improve it, and if you're interested, you can contribute to creating new applications. 

But first and foremost, I'd like to hear what @GeekMagicClock thinks about it

**Comment by GeekMagicClock** [comment]
Hi, quite good idea!  I like it, I remember some other one have simliar ideas, but your way is much easier to add.

How about add an enable or disable button and an input area to specify an img url to the Pictures APP ? And also options for image downloading interval.

**Comment by donatellosantoro** [comment]
> Hi, quite good idea! I like it, I remember some other one have simliar ideas, but your way is much easier to add.
> 
> How about add an enable or disable button and an input area to specify an img url to the Pictures APP ? And also options for image downloading interval.

Exactly! That will be enough! 
From the tecnical point of view, the webserver needs to generate a JPG/GIF, or also PNG will be supported? 

Thanks again! 

P.s: If you need a beta tester please ping me

**Comment by donatellosantoro** [comment]
Hi @GeekMagicClock 
do you have any update about this?

Thanks in advance!

**Comment by GeekMagicClock** [comment]
Hi, if you are able to program, I think this may satisfy your needs.  https://github.com/GeekMagicClock/GeekMagic-S3

For smalltv-pro project, this function needs more consideration, how to integrate into the original project without conflicting with the original functions.

**Comment by donatellosantoro** [comment]
> Hi, if you are able to program, I think this may satisfy your needs. https://github.com/GeekMagicClock/GeekMagic-S3
> 
> For smalltv-pro project, this function needs more consideration, how to integrate into the original project without conflicting with the original functions.

I will definitely take a look at that project. I see that there is an http client example, that seems exactly what I need.

However, by building the project I'm not totally sure what will be the final output. I will get a firmware to sideload to smalltv-pro? 

Thanks
